cmd/soroban-rpc/lib/preflight/src/preflight.rs

```diff
@@ -0,0 +1,286 @@
+
+    let ConfigSettingEntry::ContractCostParamsCpuInstructions(cost_params_cpu) = ledger_storage
+        .get_configuration_setting(ConfigSettingId::ContractCostParamsCpuInstructions)?
+        else {
+            bail!("get_budget_from_network_config_params(): unexpected config setting entry for ComputeV0 key");
+        };
+
+    let ConfigSettingEntry::ContractCostParamsMemoryBytes(cost_params_memory) =
+        ledger_storage.get_configuration_setting(ConfigSettingId::ContractCostParamsMemoryBytes)?
+        else {
+            bail!("get_budget_from_network_config_params(): unexpected config setting entry for ComputeV0 key");
+        };
+
+    let budget = Budget::try_from_configs(
+        compute.tx_max_instructions as u64,
+        compute.tx_memory_limit as u64,
+        cost_params_cpu,
+        cost_params_memory,
+    )
+    .context("cannot create budget from network configuration")?;
+    Ok(budget)
+}
+
+pub(crate) fn preflight_footprint_expiration_op(
+    ledger_storage: &LedgerStorage,
+    bucket_list_size: u64,
+    op_body: OperationBody,
+    footprint: LedgerFootprint,
+    current_ledger_seq: u32,
+) -> Result<PreflightResult> {
+    match op_body {
+        OperationBody::BumpFootprintExpiration(op) => preflight_bump_footprint_expiration(
+            footprint,
+            op.ledgers_to_expire,
+            ledger_storage,
+            bucket_list_size,
+            current_ledger_seq,
+        ),
+        OperationBody::RestoreFootprint(_) => preflight_restore_footprint(
+            footprint,
+            ledger_storage,
+            bucket_list_size,
+            current_ledger_seq,
+        ),
+        op => Err(anyhow!(
+            "preflight_footprint_expiration_op(): unsupported operation type {}",
+            op.name()
+        )),
+    }
+}
+
+fn preflight_bump_footprint_expiration(
+    footprint: LedgerFootprint,
+    ledgers_to_expire: u32,
+    ledger_storage: &LedgerStorage,
+    bucket_list_size: u64,
+    current_ledger_seq: u32,
+) -> Result<PreflightResult> {
+    let (transaction_data, min_fee) =
+        fees::compute_bump_footprint_exp_transaction_data_and_min_fee(
+            footprint,
+            ledgers_to_expire,
+            ledger_storage,
+            bucket_list_size,
+            current_ledger_seq,
+        )?;
+    Ok(PreflightResult {
+        auth: vec![],
+        result: None,
+        transaction_data,
+        min_fee,
+        events: vec![],
+        cpu_instructions: 0,
+        memory_bytes: 0,
+        restore_preamble: None,
+    })
+}
+
+fn preflight_restore_footprint(
+    footprint: LedgerFootprint,
+    ledger_storage: &LedgerStorage,
+    bucket_list_size: u64,
+    current_ledger_seq: u32,
+) -> Result<PreflightResult> {
+    let (transaction_data, min_fee) = fees::compute_restore_footprint_transaction_data_and_min_fee(
+        footprint,
+        ledger_storage,
+        bucket_list_size,
+        current_ledger_seq,
+    )?;
+    Ok(PreflightResult {
```

Comment:
Now a huge issue, but since this is repeated above you could also create a Default implementation and then you could do

```rust
Ok(PreflightResult {
        transaction_data,
        min_fee,
        ..Default::default()
    })
```

Reply:
That’s coming in a separate PR, thanks!!